### PR TITLE
Commenting use_label_encoder - xgboost

### PR DESCRIPTION
**Pull request by minghao51**
## Why are these changes needed?

`use_label_encoder` is deprecated since 1.7.0. (xgboost), yet it is still currently provided as part of the hyperparameter for xgboost

## Related issue number
https://github.com/microsoft/FLAML/issues/1120

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by qingyun-wu — APPROVED** [review]

**Comment by qingyun-wu** [comment]
@qingyun-wu Check benchmark option.

**Comment by qingyun-wu** [comment]
@[minghao51](https://github.com/minghao51), thank you for the contribution. We have some installation option that requires xgboost version earlier than 1.7.0.  Can you please first check the version of xgboost and only comment the line only when the version number is larger than or equal to 1.7.0? This ensure the code still works fine even when an earlier version of xgboost is used. 

Thank you!

**Comment by minghao51** [comment]
Sure

**Comment by minghao51** [comment]
Updated the PR

File: flaml/automl/model.py [hunk 1]
```diff
@@ -1509,8 +1509,12 @@ def config2params(self, config: dict) -> dict:
         if max_depth == 0:
             params["grow_policy"] = params.get("grow_policy", "lossguide")
             params["tree_method"] = params.get("tree_method", "hist")
-        # params["booster"] = params.get("booster", "gbtree")
-        params["use_label_encoder"] = params.get("use_label_encoder", False)
+            # params["booster"] = params.get("booster", "gbtree")
+            # use_label_encoder is deprecated in 1.7.0
+            from xgboost import __version__ as xgboost_version
```
Comment (qingyun-wu) [inline flaml/automl/model.py:1514]: This code block should not be under the `max_depth == 0` condition.

Comment (minghao51) [reply flaml/automl/model.py:1514]: should the import xgboost version be under the function then? or at the very start?

File: flaml/automl/model.py [hunk 2]
```diff
@@ -1504,13 +1504,17 @@ def cost_relative2lgbm(cls):
...
         params = super().config2params(config)
         max_depth = params["max_depth"] = params.get("max_depth", 0)
         if max_depth == 0:
             params["grow_policy"] = params.get("grow_policy", "lossguide")
             params["tree_method"] = params.get("tree_method", "hist")
-        # params["booster"] = params.get("booster", "gbtree")
-        params["use_label_encoder"] = params.get("use_label_encoder", False)
+            # params["booster"] = params.get("booster", "gbtree")
+            # use_label_encoder is deprecated in 1.7.
+            if xgboost_version < "1.7.0":
```
Comment (qingyun-wu) [inline flaml/automl/model.py:1516]: Sorry about the confusion. I mean all these several lines of code.

File: flaml/automl/model.py [hunk 3]
```diff
@@ -1504,13 +1504,17 @@ def cost_relative2lgbm(cls):
...
+                params["use_label_encoder"] = params.get("use_label_encoder", False)
```
Comment (qingyun-wu) [inline flaml/automl/model.py:1517]: Sorry about the confusion. I mean all these several lines of code.

Comment (qingyun-wu) [reply flaml/automl/model.py:1517]: They need to be outside of the condition `if max_depth == 0:` according to the original logic.

**Review by thinkall — APPROVED** [review]

File: flaml/automl/model.py [hunk 4]
```diff
@@ -1509,8 +1509,12 @@ def config2params(self, config: dict) -> dict:
         if max_depth == 0:
             params["grow_policy"] = params.get("grow_policy", "lossguide")
             params["tree_method"] = params.get("tree_method", "hist")
-        # params["booster"] = params.get("booster", "gbtree")
-        params["use_label_encoder"] = params.get("use_label_encoder", False)
+            # params["booster"] = params.get("booster", "gbtree")
```
Comment (thinkall) [inline flaml/automl/model.py:1512]:
```suggestion
        # params["booster"] = params.get("booster", "gbtree")

```

**Pull request merged**